content-services/content-sources-backend
# Fixes 4313: update use_latest templates after new snapshot

## Summary
- Adds a new task `update-latest-snapshot` that updates all use_latest templates to use the latest snapshot for a given repository. This task is enqueued as a dependent of every snapshot task, so that it runs after the snapshot is completed. 
- Adds support for dependencies in the tasking system. Adds the "dependencies" and "dependents" fields to TaskInfoResponse. If a parent task fails, dependent tasks are canceled. If a parent task is requeued (after failure), dependent tasks would also be requeued.
 ## Testing steps
1. Create a repository with snapshot=true
2. Create a template with use_latest=true and add the repository.
3. List the template RPMs to compare with later
4. Update the repository to change the URL
5. Wait for the snapshot and update-latest-snapshot tasks to complete
6. Fetching the task details for the snapshot task should return the UUID of the update-latest-snapshot task in the "dependents" field. Fetching the details for the update-latest-snapshot task should show the UUID of the snapshot task in the "dependencies" field.
7. List the template RPMs again, they should instead be the RPMs from the new URL
